File: MVC_GerenciadorDeCont/Scripts/cep.js

jQuery(function ($) {
    $("input[name='cep']").change(function () {
        var cep_code = $(this).val();
        if (cep_code.length <= 0) {   

            return;
        }
        $.get("https://ws.apicep.com/cep/[cepCode].json", { code: cep_code }, function (result) {
            if (result.status != 200) {
                alert(result.message || "Houve um erro desconhecido");
                return;
            }
            $("input[name='cep']").val(result.code);
            $("input[name='estado']").val(result.state);
            $("input[name='cidade']").val(result.city);
            $("input[name='bairro']").val(result.district);
            $("input[name='endereco']").val(result.address);
            $("input[name='estado']").val(result.state);
        })
    })
})

File: Business/Cep.cs
﻿using System.IO;
using System.Net;
using System.Web.Script.Serialization;

namespace Business
{
    public class Cep
    {
        public string CEP { get; set; }
        public string Endereco { get; set; }
        public string Bairro { get; set; }
        public string Cidade { get; set; }
        public string Estado { get; set; }

        public static Cep Busca (string cep)
        {
            var cepObj = new Cep();

            string com = ".json";

            var url = "https://ws.apicep.com/cep/" + cep + com;

            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
            request.AutomaticDecompression = DecompressionMethods.GZip;

            string json = string.Empty;
            using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
            using (Stream stream = response.GetResponseStream())
            using(StreamReader reader = new StreamReader(stream))
            {
                json = reader.ReadToEnd();
            }

            JavaScriptSerializer json_serializer = new JavaScriptSerializer();
            JsonCepObject cepJson = json_serializer.Deserialize<JsonCepObject>(json);

            cepObj.CEP = cepJson.code;
            cepObj.Endereco = cepJson.address;
            cepObj.Bairro = cepJson.district;
            cepObj.Cidade = cepJson.city;
            cepObj.Estado = cepJson.state;

            return cepObj;
        }

        public class JsonCepObject
        {
            public string code { get; set; }
            public string state { get; set; }
            public string city { get; set; }

            public string district { get; set; }
            public string address { get; set; }
        }
    }
}
